### PR TITLE
Feat/contact add filter

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes and refactors how ActiveCampaign ESP contact upsertion (`add_contact` method) works, and adds a filter to modify the contact data prior to sending.

Enables contact adding without lists assigned.

### How to test the changes in this Pull Request:

_Follow instructions in https://github.com/Automattic/newspack-plugin/pull/1793_

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php [hunk 1]
```diff
@@ -622,30 +622,42 @@ public function trash( $post_id ) {
...
 	 *
 	 *    @type string   $email    Contact email address.
 	 *    @type string   $name     Contact name. Optional.
 	 *    @type string[] $metadata Contact additional metadata. Optional.
 	 * }
-	 * @param string $list_id      List to add the contact to.
+	 * @param string|false $list_id      List to add the contact to.
 	 *
-	 * @return bool|WP_Error True if the contact was added or error if failed.
+	 * @return int|WP_Error Contact ID if the contact was added or error if failed.
```
Comment (miguelpeixe) [inline includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php:636]: We should standardize the returned value whether it's an array with normalized data (contact id, name, email, lists, etc) or just the contact ID.

This PR changes from the standardized `boolean` to ID for ActiveCampaign and an uncontrolled array from the ESP returned data.

File: includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php [hunk 2]
```diff
@@ -792,4 +791,22 @@ public function update_contact_lists( $email, $lists_to_add = [], $lists_to_remo
...
 	}
+
+	/**
+	 * Get contact data by email.
+	 *
+	 * @param string $email_address Email address.
+	 *
+	 * @return array|WP_Error Response or error if contact was not found.
+	 */
+	public function existing_contact_data( $email_address ) {
```
Comment (miguelpeixe) [inline includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php:802]: NIT: `contact_data` presumes it exists, no need to state that.

```suggestion
	public function contact_data( $email_address ) {
```

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php (hunk 2 above)
Comment (adekbadek) [reply includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php]: Done in d4727588a9029075dec1dbc345c376759fe034fc (and in Newspack Plugin, too)

**Comment by adekbadek** [comment]
@miguelpeixe – I've also noticed that the implementation was incomplete here. When registering w/out any newsletters chosen, the contact should be added anyway. Fixed in 28e5983da2a871a1987b8621a6562f22dca7bc3c & 1740476376ba839dafcc421349931675f002d1d6

File: includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php (hunk 1 above)
Comment (adekbadek) [reply includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php]: Done in 9dee2c39232304cf8f410bb0ecbba2209001f55c – an array with `'id'` key will always be returned.

**Review by miguelpeixe — APPROVED** [review]

**Pull request merged**